### DISCUSSION TITLE
How to install precompiled binaries

### DISCUSSION THREAD

**Discussion by parzifull** [OpenFAST/openfast#1145] [General]
Hello,

I am struggling to find an older version of the software so if anyone has it or is able to point me in the direction of v2.0 that would be amazing. I am trying to download any version of Open fast but appear to be really struggling to get the new version up and running.

**Comment by bjonkman** [comment]
You can find all of the releases on the OpenFAST GitHub releases page: https://github.com/OpenFAST/openfast/releases. In particular, v2.0.0 is here: https://github.com/OpenFAST/openfast/releases/tag/v2.0.0.

**Reply by rafmudaf** [reply]
Additionally, on Linux and macOS you can install OpenFAST from v2.6 to the latest version with conda:

```bash
conda install -c conda-forge openfast
```

**Reply by PaulKSmith** [reply]
Hello @rafmudaf 

I need to install openfast v3.3.0 in anaconda, however, `conda install -c conda-forge openfast=v3.3.0` does not work. Am I missing something here? Thanks in advance.

**Reply by rafmudaf** [reply]
Hi @PaulKSmith please elaborate on "does not work" and provide any details that you think would be helpful for us to understand your situation.

**Reply by PaulKSmith** [reply]
@rafmudaf Here it is. Is it available in different channel?

```
(rosco-2.9) C:\>conda install -c conda-forge openfast=v3.3.0
Collecting package metadata (current_repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - openfast=v3.3.0

Current channels:

  - https://conda.anaconda.org/conda-forge/win-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

**Reply by rafmudaf** [reply]
@PaulKSmith [Here's the list](https://anaconda.org/conda-forge/openfast/files?sort=basename&sort_order=desc) of files distributed through conda (via conda-forge). For Windows, v3.3.0 is only available with the `rc` label. Being quite old, I recommend to not use that one as the process for distributing with conda for Windows was recently overhauled. Instead, I suggest to use v3.5.1, v3.5.2, or the `dev` label.

**Reply by PaulKSmith** [reply]
@rafmudaf Thanks for your suggestions

**Comment by rafmudaf** [comment]
Hi @parzifull regarding the latest version, what have you tried and what is the problem?

**Reply by parzifull** [reply]
I have been trying to follow the guide that is linked on the first page but it seems to be for v2. I downloaded everything and have had no luck with the newest version. This is my first time trying to compile this sort of software so I'm struggling even with the guide.

**Reply by rafmudaf** [reply]
If you can share some details, we can give you better guidance.

- Which operating system are you using?
- Share a link to "guide that is linked on the first page" that you are reading.
- Are you sure you need to [compile](https://www.quora.com/What-does-it-mean-to-compile-a-computer-program) OpenFAST or would downloading a precompiled executable be easier for your case? If you have not changed the source code and are not using an usual computer or operating system, you could simply download a precompiled executable.

**Reply by parzifull** [reply]
I was unaware there was a precompiled executable and would love to just use that instead. I am running Windows and the link to the guide is here: https://openfast.readthedocs.io/en/main/source/install/index.html#visual-studio-solution-for-windows

Thank you so much

**Reply by rafmudaf** [reply]
Use the section just before that: https://openfast.readthedocs.io/en/main/source/install/index.html#download-binaries
Specifically, look at the GitHub Releases. This is also what @bjonkman described in her response.

**Reply by parzifull** [reply]
Okay awesome! Would I also need Matlab for this precompiled version then?

**Reply by rafmudaf** [reply]
Only if you will be using Simulink for controller development. If not, you simply use the OpenFAST executable to run a simulation like this:

```bash
openfast.exe input_file.fst
```

Of course, change the names of the executable and input file path so that it matches your situation.

**Reply by parzifull** [reply]
Okay, I successfully clone the repository but the files are now reading as (incompatible). I'm wondering if I forgot to install an extra feature at some point?

**Reply by rafmudaf** [reply]
@parzifull please provide more details. You can copy the text from the command line, list specific commands that you used, etc.

**Reply by parzifull** [reply]
Unsupported
This version of Visual Studio is unable to open the following projects. The project types may not be installed or this version of Visual Studio may not support them. 
For more information on enabling these project types or otherwise migrating your assets, please see the details in the "Migration Report" displayed after clicking OK.
	 - AeroDyn_Driver, "C:\Users\flagu\Documents\openfastnew\vs-build\AeroDyn\AeroDyn_Driver.vfproj"


No changes required
These projects can be opened in Visual Studio 2015, Visual Studio 2013, Visual Studio 2012, and Visual Studio 2010 SP1 without changing them.
	 - FAST_Registry, "C:\Users\flagu\Documents\openfastnew\vs-build\Registry\FAST_Registry.vcxproj"
	 - AeroDyn_Driver, "C:\Users\flagu\Documents\openfastnew\vs-build\AeroDyn\AeroDyn_Driver.sln"

**Reply by parzifull** [reply]
^ this is the message i receive when I attempt to open any of the files from the repository such as tests

**Comment by parzifull** [comment]
Okay, I will give it a try I just need to finish installing Microsoft visual and Intel Fortran.

**Reply by parzifull** [reply]
Okay I think everything is downloaded correctly but now when I compile I get an error on line 18 that reads : project(OpenFAST CXX C Fortran)

**Comment by bjonkman** [comment]
When you use the pre-compiled binaries (executables), you do not need to install Visual Studio or Intel Fortran because you do not need to compile them yourself.

Errors about not being able to open projects in Visual Studio are generally caused by either
1. not clicking the box to install integrations into Visual Studio when you are installing Intel Fortran (or the oneAPI Base and HPC toolkit), or
2. using a version of Visual Studio that is not supported by Intel (for example, see https://github.com/OpenFAST/openfast/issues/1133).
